config/routes.rb

```diff
@@ -1,5 +1,6 @@
 Rails.application.routes.draw do
-  root to: 'texts#new'
+  root to: 'snippets#new'
 
-  resources :texts, only: %i[create show], param: :token
+  get '/text/:token', to: redirect('/t/%{token}')
+  resources :t, only: %i[create show], param: :token, controller: :snippets, as: :snippets
```

Comment:
`t` in routes stand for `text`, not `token`, I don't insist, but I suggest change it to `s` so that path is:

```
https://droptext.ru/s/123
```